chacha20poly1305/Cargo.toml

```diff
@@ -28,13 +28,16 @@ zeroize = { version = "=1.3", default-features = false }
 aead = { version = "0.4", features = ["dev"], default-features = false }
 
 [features]
-default = ["alloc", "chacha20", "xchacha20poly1305"]
+default = ["alloc", "chacha20", "xchacha20"]
 std = ["aead/std", "alloc"]
 alloc = ["aead/alloc"]
 heapless = ["aead/heapless"]
-reduced-round = ["chacha20"]
 stream = ["aead/stream"]
-xchacha20poly1305 = ["chacha20/xchacha"]
+xchacha20 = ["chacha20/xchacha"]
```

Comment:
I feel a bit of explanation is needed as to why there's a `chacha20` feature in the first place, as I feel there doesn't need to be an `xchacha20` feature.

This feature activates the `chacha20` *crate*.

The reason why it's optional is because the `c2-chacha` crate provided a faster implementation and also impl'd the RustCrypto traits, so people who really cared about performance could swap it in instead, and disable the `chacha20` crate.

With https://github.com/RustCrypto/stream-ciphers/pull/261 the gap has narrowed considerably (ty @str4d).

I think in the next breaking release we can make `chacha20` a mandatory dependency again, which would eliminate the need for this feature.

All that said, it seems like `xchacha20poly1305` and `xchacha20-reduced-round` could both activate `chacha20/xchacha`. And perhaps it would also make sense for simplicity's sake to get rid of `chacha20/xchacha` and always include the `XChaCha` family as the code size really isn't that big.

Anyway this is all a long winded way of saying it feels like there are too many features here and I take responsibility for at least part of that.